avro/src/main/java/com/fasterxml/jackson/dataformat/avro/schema/RecordVisitor.java

```diff
@@ -27,16 +37,31 @@ public RecordVisitor(SerializerProvider p, JavaType type, DefinedSchemas schemas
         super(p);
         _type = type;
         _schemas = schemas;
-        _avroSchema = Schema.createRecord(AvroSchemaHelper.getName(type),
-                "Schema for "+type.toCanonical(),
-                AvroSchemaHelper.getNamespace(type), false);
+        // Check if the schema for this record is overridden
+        AnnotatedClass ac = getProvider().getConfig().introspectDirectClassAnnotations(_type).getClassInfo();
+        AvroSchema ann = ac.getAnnotation(AvroSchema.class);
+        if (ann != null) {
+            Schema.Parser parser = new Schema.Parser();
```

Comment:
Would make sense to encapsulate in the avro schema helper (I know it's short snippet but still... may want to redo schema parser in future).